clash-prelude/src/Clash/Signal/Internal.hs

```diff
@@ -439,6 +451,33 @@ type HasDefinedInitialValues (dom :: Domain) =
 type DomainResetPolarity (dom :: Domain) =
   DomainConfigurationResetPolarity (KnownConf dom)
 
+-- * Time rep
+-- | Gets time in 'Picoseconds' from time in 'Seconds'.
+type Seconds      (s  :: Nat) = Milliseconds (1000 * s)
+-- | Gets time in 'Picoseconds' from time in 'Milliseconds'.
+type Milliseconds (ms :: Nat) = Microseconds (1000 * ms)
+-- | Gets time in 'Picoseconds' from time in 'Microseconds'.
+type Microseconds (us :: Nat) = Nanoseconds  (1000 * us)
+-- | Gets time in 'Picoseconds' from time in 'Nanoseconds'.
+type Nanoseconds  (ns :: Nat) = Picoseconds  (1000 * ns)
+-- | Gets time in 'Picoseconds' from time in 'Picoseconds', essentially 'id'.
+type Picoseconds  (ps :: Nat) = ps
+
+-- | Converts a frequency in Hz to a period in picoseconds. This might lead to rounding
+-- errors.
+type HzToPeriod hz = Seconds 1 `Div` hz
```

Comment:
```suggestion
type HzToPeriod (hz :: Nat) = Seconds 1 `Div` hz
```

We're not really completely consistent with this, but we usually do do it. As you do with most tyargs here as well.

And it "leaks" to the Haddock, it's directly user visible.